### PR TITLE
feat: add `disableTransition` option and set default to `true`

### PR DESCRIPTION
**Pull request by zcf0508**
Reference: 

https://vueuse.org/core/useColorMode/
https://paco.me/writing/disable-theme-transitions

### CODE REVIEW

File: src/script.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ addColorScheme(value)
 w['<%= options.globalName %>'] = {
   preference,
   value,
+  disableTransition: '<%= options.disableTransition %>',
```
Comment (Atinux) [inline src/script.ts:23]: Why forcing it to be a string here?

Comment (zcf0508) [reply src/script.ts]: I have no idea how to use `boolean` type when `addTemplate` processes the file. 

Perhaps `'<%= options.disableTransition %>' === 'true'` is an optional solution?

**Review by antfu — APPROVED** [review]

**Comment by antfu** [comment]
I am a bit hesitate if this should be enabled by default. Would love to have some feedback from the community.

**Comment by antfu** [comment]
I think it might be better to have it default to `false` for now, so we can move forward. Later  we could revise the defaults on the next major

**Comment by zcf0508** [comment]
> I think it might be better to have it default to `false` for now, so we can move forward. Later we could revise the defaults on the next major

OK, I think it is better too.

**Comment by hacknug** [comment]
Perhaps default could come from [`@media/prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion) and then config allows you to force enable/disable.

In case it counts, I'd prefer transitions to be disabled by default.

**Comment by Stozen-spb** [comment]
I'm looking forward for this feature

File: src/script.ts [hunk 2]
```diff
@@ -20,6 +20,8 @@
   w['<%= options.globalName %>'] = {
     preference,
     value,
+    // @ts-ignore
+    disableTransition: '<%= options.disableTransition %>' === 'true',
```
[the remark this answers is not in the document]
Comment (zcf0508) [reply src/script.ts]: I'm sorry that I don't quite understand what you say `tree-shake`. I think this is just an injected file.

[the remark this answers is not in the document]
Comment (danielroe) [reply src/script.ts]: updated!

[the remark this answers is not in the document]
Comment (zcf0508) [reply src/script.ts]: Awesome! 👍

**Pull request merged**